File: vseros/2016-17/okr/2.cpp
#include <iostream>

using namespace std;

int gcd(int a, int b) { return (b) ? gcd(b, a % b) : a; }

int main()
{
	int a, b, c, d;
	std::cin >> a >> b >> c >> d;
	int i = 0;
	while (a * d < c * b)
	{
		++a;
		++b;
		int vGcd = gcd(a, b);
		a /= vGcd;
		b /= vGcd;
		++i;
	}
	cout << ((a * d > c * b) ? 0 : i) << endl;
	return 0;
}

File: vseros/2016-17/okr/2.py
def gcd(x, y):
    return y if x % y == 0 else gcd(y, x % y)

a = int(input())
b = int(input())
c = int(input())
d = int(input())
k = 0
while a / b < c / d:
    a += 1
    b += 1
    n = gcd(b, a)
    a /= n
    b /= n
    k += 1
if a / b == c / d:
    print(k)
else:
    print(0)
